python/pyspark/sql/dataframe.py

```diff
@@ -390,6 +391,18 @@ def collect(self):
             port = self._jdf.collectToPython()
         return list(_load_from_socket(port, BatchedSerializer(PickleSerializer())))
 
+    @since(2.2)
+    def collectAsArrow(self):
```

Comment:
As we chatted it might make sense to make this an internal method for now (`_collectAsArrow`).